lib/sinatra/base.rb

```diff
@@ -1834,7 +1834,7 @@ def self.force_encoding(data, *) data end
 
     reset!
 
-    set :environment, (ENV['RACK_ENV'] || :development).to_sym
+    set :environment, (ENV['SINATRA_ENV'] || :development).to_sym
```

Comment:
Sorry, I missed your note about this being for 2.0. Perhaps my suggestion would enable this to be merged to the main branch?